### PR TITLE
Transcripts fix support for alternative format types

### PR DESCRIPTION
**Pull request by SergioEstevao**
| 📘 Part of: #2185 |  
|:---:|

Fixes # 

This PR updates the code so transcript format can have more than one valid mime type that match them.

## To test

1. Start the app
2. Open the following podcast: https://pca.st/episode/2cb385e0-e8f2-4b5d-be50-69b203630174
3. Play it and open the full screen player
4. Tap on the Transcript icon
5. Check if the transcript shows correctly

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/59">7.73 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/TranscriptFormat.swift [hunk 1]
```diff
@@ -36,6 +49,11 @@ public enum TranscriptFormat: String {
 
 extension Episode.Metadata.Transcript {
     public var transcriptFormat: TranscriptFormat? {
-        return TranscriptFormat(rawValue: self.type)
+        for format in TranscriptFormat.allCases {
```
Comment (danielebogo) [inline podcasts/TranscriptFormat.swift]: Nit:
I think you can either filter the for using `where`
```
for format in TranscriptFormat.allCases where format.possibleTypes.contains(type) {
            return format
        }
        return nil
```

or just `return TranscriptFormat.allCases.first { $0.possibleTypes.contains(type) }`

**Review by danielebogo — APPROVED** [review]
@SergioEstevao I tested the episode and it works!
I left one nit only.

File: podcasts/TranscriptFormat.swift (hunk 1 above)
Comment (SergioEstevao) [reply podcasts/TranscriptFormat.swift]: Done here: https://github.com/Automattic/pocket-casts-ios/pull/2193/commits/29f14f6563dfed79ab621953fb41ea63b5aa0e8a

**Pull request merged**